consumer/src/main/kotlin/streams/kafka/KafkaEventSink.kt

```diff
@@ -153,7 +153,11 @@ open class KafkaAutoCommitEventConsumer(private val config: KafkaSinkConfigurati
         if (!records.isEmpty) {
             try {
                 this.topics.forEach { topic ->
-                    action(topic, records.records(topic).map { JSONUtils.readValue<Any>(it.value()) })
+                    val topicRecords = records.records(topic)
+                    if (!topicRecords.iterator().hasNext()) {
```

Comment:
can't you just do the . action in the if, and not have the lambda return.
removes one negation and makes the code simpler.

Reply:
also it's odd as it should just have called the action with an empty list ??

Reply:
In this case the check is a plus, we can omitted it, it's just because I don't want pass empty lists.
Wdyt if, as you suggested in the next comment, I do `topics.map.filter.foreach` also here?

Reply:
yes, makes sense. easier to reason about too.